app/controllers/commodities_controller.rb

```diff
@@ -38,10 +40,14 @@ def commodity
   end
 
   def xi_commodity
-    @commodities[:xi]
+    commodities[:xi]
   end
 
   def uk_commodity
-    @commodities[:uk]
+    commodities[:uk]
+  end
+
+  def set_session
+    session[:commodity_code] = commodity.code if commodity
```

Comment:
I think if we don't have a `commodity` we'll have a cascade of many other problems. Probably don't need the if, here, as is defensive